msal/application.py

```diff
@@ -428,9 +428,10 @@ def acquire_token_silent(
             if result.get("access_token"):
                 return result
             if error_response:
-                if result.get("suberror") in set(["bad_token", "token_expired", "client_mismatch"]):
-                    result = None
-                return Error(result['error'], result['error_description'], result['suberror'])
+                classification = result.get("suberror")
+                if classification:
+                    if classification not in set(["bad_token", "token_expired", "client_mismatch"]):
+                        response = Error(result['error'], result['error_description'], result['suberror'])
```

Comment:
I thought in last Tue's discussion, Henrik and us agreed on returning a dictionary as the return type?